packages/runtime/container-runtime/src/summary/documentSchema.ts

```diff
@@ -480,6 +519,11 @@ export class DocumentsSchemaController {
 		return schema;
 	}
 
+	/**
+	 * Called by Container runtime whenever it is about to send some op.
+	 * It gives opportunity for controller to issue its own ops - we do not want to send ops if there are no local changes in document.
+	 * @param send - delegate that controller can use to send its own op, if it needs to.
+	 */
```

Comment:
As per my comment on the test, maybe this should be called something like `maybeSendSchemaChangeMessage` or `maybeUpdateSchema` to make it clear that its not just called in response to a message being sent but it can actually sent its own message too.

Reply:
I'll revert the flow - instead of taking callback, it will return a message (optional) to send.